emaust

```diff
@@ -0,0 +1,101 @@
+drive_history = {
+  DR0001: [
+    {cost: 10, rating: 3, id_number: "RD0003", date: "3rd Feb 2016"},
+    {cost: 30, rating: 4, id_number: "RD0015", date: "3rd Feb 2016"}, 
+    {cost: 45, rating: 2, id_number: "RD0003", date: "5th Feb 2016"}
+  ],
+  DR0002: [
+    {cost: 25, rating: 5, id_number: "RD0073", date: "3rd Feb 2016"},
+    {cost: 15, rating: 1, id_number: "RD0013", date: "4th Feb 2016"}, 
+    {cost: 35, rating: 3, id_number: "RD0066", date: "5th Feb 2016"}
+  ],
+  DR0003: [
+    {cost: 05, rating: 5, id_number: "RD0066", date: "4th Feb 2016"},
+    {cost: 50, rating: 2, id_number: "RD0003", date: "5th Feb 2016"}
+  ],
+  DR0004: [
+    {cost: 05, rating: 5, id_number: "RD0022", date: "3rd Feb 2016"},
+    {cost: 10, rating: 4, id_number: "RD0022", date: "4th Feb 2016"}, 
+    {cost: 20, rating: 5, id_number: "RD0073", date: "5th Feb 2016"}
+  ]
+}
+
+puts " *** Total Rides Per Driver:"
+rides_completed = []
+drive_history.each do |driver, key|
+  puts "For driver #{driver}, the number of rides given was: #{key.length}"
+  drive_number = key.length
+  rides_completed << drive_number
+end
+
+
+def profit_per_driver(drive_history, driver)
+  cost_array = drive_history[driver].map do |ride|
+    cost_each = ride[:cost]
+  end
+end
+
+
+prof_one = profit_per_driver(drive_history,:DR0001).sum
```

Comment:
Considering this is called profit_per_driver, it'd be preferable that it found and returned the sum of the profit rather than an array that still needs to be summed.